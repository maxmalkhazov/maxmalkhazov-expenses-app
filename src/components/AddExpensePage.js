import React from "react";
import ReactDOM from "react-dom";

const AddExpensePage = () => (
	<div>
		This is from my AddExpense component!
	</div>
);

export default AddExpensePage;