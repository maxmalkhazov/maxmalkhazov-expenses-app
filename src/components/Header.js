import React from "react";
import ReactDOM from "react-dom";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
	<header>
		<NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
		<NavLink to="/create" activeClassName="is-active">Create</NavLink>
		<NavLink to="/help" activeClassName="is-active">Help</NavLink>
		<h1>Expenses-App</h1>
		<button onClick={startLogout}>Logout</button>
	</header>
);

const mapDispatchToProps = (dispatch) => ({
	startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);