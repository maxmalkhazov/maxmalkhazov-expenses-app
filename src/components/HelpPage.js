import React from "react";
import ReactDOM from "react-dom";

const HelpPage = () => (
	<div>
		This is from my Help page component
	</div>
);

export default HelpPage;