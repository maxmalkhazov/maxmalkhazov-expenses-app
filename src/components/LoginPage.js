import React from "react";
import ReactDOM from "react-dom";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const LoginPage = ({ startLogin }) => (
	<div>
		<button onClick={startLogin}>Login</button>
	</div>
);

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);