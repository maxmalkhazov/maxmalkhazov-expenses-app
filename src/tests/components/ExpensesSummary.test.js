import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should correctly render ExpensesSummary with 1 expense", () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={125} />);
	expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpensesSummary with mutiple expenses", () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={10} expensesTotal={500} />);
	expect(wrapper).toMatchSnapshot();
});