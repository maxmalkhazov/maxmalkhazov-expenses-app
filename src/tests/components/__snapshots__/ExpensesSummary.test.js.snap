// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.25
    .
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with mutiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    10
     
    expenses
     totalling 
    $5.00
    .
  </h1>
</div>
`;
